src/NumericalAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -0,0 +1,416 @@
+                   vars_dest_all_times) noexcept {
+        auto& vars_dest = vars_dest_all_times->at(temporal_id);
+        // Here we assume that vars_dest has been allocated to the correct
+        // size (but could contain garbage, since below we are filling it).
+        const size_t npts_dest = vars_dest.number_of_grid_points();
+        const size_t nvars = vars_dest.number_of_independent_components;
+        for (size_t j = 0; j < global_offsets.size(); ++j) {
+          const size_t npts_src = global_offsets[j].size();
+          for (size_t i = 0; i < npts_src; ++i) {
+            // If a point is on the boundary of two (or more)
+            // elements, it is possible that we have received data
+            // for this point from more than one Interpolator.
+            // This will rarely occur, but it does occur, e.g. when
+            // a point is exactly on some symmetry
+            // boundary (such as the x-y plane) and this symmetry
+            // boundary is exactly the boundary between two
+            // elements.  If this happens, we accept the first
+            // duplicated point, and we ignore subsequent
+            // duplicated points.  The points are easy to keep track
+            // of because global_offsets uniquely identifies them.
+            if ((*indices_of_filled)[temporal_id]
+                    .insert(global_offsets[j][i])
+                    .second) {
+              for (size_t v = 0; v < nvars; ++v) {
+                // clang-tidy: no pointer arithmetic
+                vars_dest.data()[global_offsets[j][i] +    // NOLINT
+                                 v * npts_dest] =          // NOLINT
+                    vars_src[j].data()[i + v * npts_src];  // NOLINT
+              }
+            }
+          }
+        }
+      });
+}
+
+
+/// Computes the block logical coordinates of an InterpolationTarget.
+///
+/// get_block_logical_coords is called by an Action of InterpolationTarget.
+///
+/// Currently two Actions call get_block_logical_coords:
+///   SendPointsToInterpolator (called by AddTemporalIdsToInterpolationTarget
+///                             and by FindApparentHorizon)
+///   InterpolationTargetVarsFromElement (called by DgElementArray)
+template <typename InterpolationTargetTag, typename DbTags,
+          typename Metavariables, typename TemporalId>
+auto get_block_logical_coords(const db::DataBox<DbTags>& box,
+                              Parallel::ConstGlobalCache<Metavariables>& cache,
+                              const TemporalId& temporal_id) noexcept {
+  const auto& domain =
+      db::get<domain::Tags::Domain<Metavariables::volume_dim>>(box);
+  return block_logical_coordinates(
+      domain, InterpolationTargetTag::compute_target_points::points(
+                  box, cache, temporal_id));
+}
+
+/// Adds the supplied block logical coordinates to the
+/// InterpolationTarget's internal DataBox, and initializes
```

Comment:
This is the comment that cause my confusion. As far as I can tell from your answer above, and from the code below, the coords are not actually added to the Databox?

Reply:
Ah, ok I will fix that!